### PR TITLE
Add SQL server resources

### PR DESCRIPTION
**Pull request by windoze**
Add SQL server resources.
Predefined tables required by the registry is included as a BACPAC file, which I have to put in another storage account because I don't have access to the subscription used by Feathr publications. But it's just the default value and can be overridden on deployment.

NOTE: This PR is to **add resources only**, Richin is still working at the same place to provide a full template with all components fully integrated, along with the document.

### CODE REVIEW

File: docs/how-to-guides/azure_resource_provision.json [hunk 1]
```diff
@@ -313,6 +359,59 @@
         "principalId": "[parameters('principalId')]",
         "scope": "[resourceGroup().id]"
       }
+    },
+    {
+      "type": "Microsoft.Sql/servers",
+      "apiVersion": "2021-02-01-preview",
```
Comment (xiaoyongzhu) [inline docs/how-to-guides/azure_resource_provision.json:365]: use latest possible version?

File: docs/how-to-guides/azure_resource_provision.json [hunk 2]
```diff
@@ -313,6 +359,59 @@
...
+      "name": "[concat(string(parameters('databaseServerName')), '/', string(parameters('databaseName')))]",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Sql/servers/', parameters('databaseServerName'))]"
+      ],
+      "resources": [
+        {
+          "type": "extensions",
+          "apiVersion": "2014-04-01",
```
Comment (xiaoyongzhu) [inline docs/how-to-guides/azure_resource_provision.json:400]: use latest possible version?

**Review by Yuqing-cat — DISMISSED** [review]

File: docs/how-to-guides/azure_resource_provision.json [hunk 3]
```diff
@@ -39,6 +39,52 @@
...
+    },
+    "storageAccountKey": {
+      "type": "string",
+      "metadata": {
+        "description": "Specifies the key of the storage account where the BACPAC file is stored."
+      }
+    },
+    "bacpacUrl": {
+      "type": "string",
+      "defaultValue": "https://xchfeathrtest4sto.blob.core.windows.net/public/feathr-2022-6-16-18-16.bacpac",
```
Comment (jainr) [inline docs/how-to-guides/azure_resource_provision.json:84]: Does this mean we need to recreate the backup each time there is change in SQL script? Is there an automated way to do it?

Comment (windoze) [reply docs/how-to-guides/azure_resource_provision.json:84]: It's true but I don't think we need to update SQL script very often, even I did it manually still I believe it's possible to create this bacpac file automatically in our CI process, but I think we can put this item on our task list and keep focusing on higher priority tasks for now.

Comment (windoze) [reply docs/how-to-guides/azure_resource_provision.json:84]: No we can't, ARM template doesn't support execution of SQL script.

File: docs/how-to-guides/azure_resource_provision.json [hunk 4]
```diff
@@ -39,6 +39,52 @@
...
+      "metadata": {
+        "description": "Specifies the URL of the BACPAC file."
```
Comment (xiaoyongzhu) [inline docs/how-to-guides/azure_resource_provision.json:86]: Maybe specify what's this BACPAC file/SQL Database is for? So users know how to deal with it or if they even want it or not

File: docs/how-to-guides/azure_resource_provision.json [hunk 5]
```diff
@@ -39,6 +39,52 @@
       "metadata": {
         "description": "Whether or not to deploy eventhub provision script"
       }
+    },
+    "databaseServerName": {
+      "type": "string",
+      "defaultValue": "[concat('server-', uniqueString(resourceGroup().id, deployment().name))]",
```
Comment (xiaoyongzhu) [inline docs/how-to-guides/azure_resource_provision.json:45]: It's not showing up in the ARM template correctly (maybe I did something wrong on my end). can you attach a screenshot on the ARM template page?

File: docs/how-to-guides/azure_resource_provision.json [hunk 6]
```diff
@@ -313,6 +359,59 @@
...
+          ],
+          "properties": {
+            "startIpAddress": "0.0.0.0",
+            "endIpAddress": "0.0.0.0"
+          }
+        }
+      ]
+    },
+    {
+      "type": "Microsoft.Sql/servers/databases",
```
Comment (blrchen) [inline docs/how-to-guides/azure_resource_provision.json:390]: How should container app access this sql db, use managed identity or db connection string

File: docs/how-to-guides/azure_resource_provision.json (hunk 5 above)
Comment (windoze) [reply docs/how-to-guides/azure_resource_provision.json:45]: image
It shows correctly here

File: docs/how-to-guides/azure_resource_provision.json (hunk 6 above)
Comment (windoze) [reply docs/how-to-guides/azure_resource_provision.json:390]: The final connection string can be generated from the template, it should be:
```
Server=tcp:${databaseServerName}.database.windows.net,1433;Initial Catalog=${databaseName};Persist Security Info=False;User ID=${adminUser}@${databaseServerName};Password=${adminPassword};MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;
```

**Review by jainr — APPROVED** [review]

**Review by blrchen — APPROVED** [review]

**Comment by windoze** [comment]
@jainr @blrchen merge conflict resolved, but you'll need to approve again before merging

**Review by blrchen — APPROVED** [review]

**Review by jainr — APPROVED** [review]

**Pull request merged**